### ISSUE TITLE
Use of Server-only-routes?

### ISSUE THREAD

**Issue by whoisandy** [ryanflorence/react-project#24]
Considering a universal react redux application to be specific, what would be the use case of server-only-routes?  These routes could be implemented as REST API endpoints on the server side and can be used to fetch JSON data via redux right? I mean my question is do we actually need the server-only-routes if we are actually trying to build a universal react redux app?

**Comment by KraigWalker** [comment]
Some use cases that stick out in my mind involve XML output or Transnational Email Templating. Something I've been curious about has been a "send to friend" function where an email of a piece of content from an app or site can be put into an html email friendly layout and pushed to a mail service (either internal or a 3rd party api) 

That's quite a niche use case. Something more general could be Rendering an RSS Feed with React. In addition to RSS and Atom, Google have a number of different XML feed types that they specify for different purposes (i.e. sitemaps, image sitemaps, social activity feeds, google news feeds etc) 

Server only routes would be is a really easy way of delivering content in these sort of scenarios - where our "user" isn't necessarily a person using at our app on their device.

**Comment by ryanflorence** [comment]
For me personally, on https://reactjs-training.com we process credit card transactions through stripe. I can't put my stripe secret key in somebody's browser, and the stripe api won't even work there if I tried.

Without these server routes I'd need to have an entirely new app to process credit card transactions. Seems silly.

If I needed to do 30 other things that were server-only, I'd probably spin up a new app and not use the server routes in this project at all.

So, in other words "because we can." :P

**Comment by ryanflorence** [comment]
> I mean my question is do we actually need the server-only-routes if we are actually trying to build a universal react redux app?

Eventually you've got a persistence store somewhere outside of the browser, it can be a new service or you can go ahead and just build it w/ the same infrastructure this project gives you. Few apps, from top to bottom, can be universal.

**Comment by whoisandy** [comment]
Sounds good, was trying to get how exactly ssr would be useful, and the examples above pretty much explain it. :)
